### PR TITLE
chore: add support for peer updates to tailnet.configMaps

### PR DESCRIPTION
**Pull request by spikecurtis**
Adds support to configMaps to handle peer updates including lost and disconnected peers

### CODE REVIEW

**Comment by spikecurtis** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/11487?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #11398** <a href="https://app.graphite.dev/github/pr/coder/coder/11398?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11399** <a href="https://app.graphite.dev/github/pr/coder/coder/11399?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11400** <a href="https://app.graphite.dev/github/pr/coder/coder/11400?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #11487** <a href="https://app.graphite.dev/github/pr/coder/coder/11487?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/coder/coder/11487?utm_source=stack-comment).

File: tailnet/configmaps.go [hunk 1]
```diff
@@ -260,11 +275,208 @@ func (c *configMaps) filterLocked() *filter.Filter {
...
+			node.Endpoints = nil
+		}
+	}
+	switch {
+	case !ok && update.Kind == proto.CoordinateResponse_PeerUpdate_NODE:
+		// new!
+		var lastHandshake time.Time
+		if ps, ok := status.Peer[node.Key]; ok {
+			lastHandshake = ps.LastHandshake
+		}
```
Comment (coadler) [inline tailnet/configmaps.go:355]: Is there a situation a peer would be missing from the `configMaps` peer list but still be in the engine's peer list?

File: tailnet/configmaps_internal_test.go [hunk 2]
```diff
@@ -89,20 +96,385 @@ func TestConfigMaps_setAddresses_same(t *testing.T) {
...
+	nodePrivateKey := key.NewNode()
+	nodeID := tailcfg.NodeID(5)
+	discoKey := key.NewDisco()
+	uut := newConfigMaps(logger, fEng, nodeID, nodePrivateKey, discoKey.Public(), nil)
+	defer uut.close()
+
+	// Then: we don't configure
+	requireNeverConfigures(ctx, t, uut)
+
+	p1ID := uuid.MustParse("10000000-0000-0000-0000-000000000000")
```
Comment (coadler) [inline tailnet/configmaps_internal_test.go:195]: A nice shorthand for this is `uuid.UUID{1}`

**Review by coadler — APPROVED** [review]

File: tailnet/configmaps.go (hunk 1 above)
Comment (spikecurtis) [reply tailnet/configmaps.go:355]: Yeah, immediately after receiving a DISCONNECTED update, we remove it from configMaps. Then until the next update it'll be in the engine but not configMaps.  Probably an edge case for us to DISCONNECT and then immediately get a new NODE update, though.

**Comment by spikecurtis** [comment]
### Merge activity

* **Jan 11, 12:11 AM**: @spikecurtis merged this pull request with [Graphite](https://app.graphite.dev/github/pr/coder/coder/11487).

**Pull request merged**